plugin/trino-opensearch/src/test/java/io/trino/plugin/opensearch/BaseOpenSearchConnectorTest.java

```diff
@@ -1883,6 +1890,369 @@ public void testQueryTableFunction()
+    @Override // Override as connector doesn't support creating tables.
+    public void testProjectionWithCaseSensitiveField()
+    {
+        String tableName = "test_projection_with_case_sensitive_field";
+        String properties =
+                """
+                        {
+                            "properties": {
+                                "id": {
+                                    "type": "integer"
+                                },
+                                "a": {
+                                    "properties": {
+                                        "UPPER_CASE": {
+                                            "type": "integer"
+                                        },
+                                        "lower_case": {
+                                            "type": "integer"
+                                        },
+                                        "MiXeD_cAsE": {
+                                            "type": "integer"
+                                        }
+                                    }
+                                }
+                            }
+                        }""";
+
+        try {
+            createIndex(tableName, properties);
+            index(tableName, ImmutableMap.<String, Object>builder()
+                    .put("id", 1L)
+                    .put("a", ImmutableMap.<String, Object>builder()
+                            .put("UPPER_CASE", 2)
+                            .put("lower_case", 3)
+                            .put("MiXeD_cAsE", 4)
+                            .buildOrThrow())
+                    .buildOrThrow());
+            index(tableName, ImmutableMap.<String, Object>builder()
+                    .put("id", 5L)
+                    .put("a", ImmutableMap.<String, Object>builder()
+                            .put("UPPER_CASE", 6)
+                            .put("lower_case", 7)
+                            .put("MiXeD_cAsE", 8)
+                            .buildOrThrow())
+                    .buildOrThrow());
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        String expected = "VALUES (2, 3, 4), (6, 7, 8)";
+        assertThat(query("SELECT a.UPPER_CASE, a.lower_case, a.MiXeD_cAsE FROM " + tableName))
+                .matches(expected)
+                .isFullyPushedDown();
+        assertThat(query("SELECT a.upper_case, a.lower_case, a.mixed_case FROM " + tableName))
+                .matches(expected)
+                .isFullyPushedDown();
+        assertThat(query("SELECT a.UPPER_CASE, a.LOWER_CASE, a.MIXED_CASE FROM " + tableName))
+                .matches(expected)
+                .isFullyPushedDown();
+    }
+
+    @Test
+    @Override // Override as connector doesn't support creating tables.
+    public void testProjectionPushdownMultipleRows()
+    {
+        String tableName = "test_projection_pushdown_multiple_rows";
+        String properties =
```

Comment:
is there a test for type "object" ?

Reply:
you mean native object type? there is a test for nested structure appears as ROW in Trino

Reply:
ya native `type: object` type